src/data/cam_thermo.F90

```diff
@@ -1554,28 +1577,32 @@ end subroutine cam_thermo_calc_kappav_2hd
    !
    !***************************************************************************
    !
-   subroutine get_hydrostatic_energy_1hd(tracer, pdel, cp_or_cv, U, V, T,     &
-        vcoord, ps, phis, z_mid, dycore_idx, qidx, te, se, ke,                &
-        wv, H2O, liq, ice)
+   subroutine get_hydrostatic_energy_1hd(tracer, moist_mixing_ratio, pdel_in, &
+        cp_or_cv, U, V, T, vcoord, ptop, phis, z_mid, dycore_idx, qidx,       &
+        te, se, po, ke, wv, H2O, liq, ice)
 
+      use cam_thermo_formula, only: ENERGY_FORMULA_DYCORE_FV, ENERGY_FORMULA_DYCORE_SE, ENERGY_FORMULA_DYCORE_MPAS
       use cam_logfile,     only: iulog
-      use dyn_tests_utils, only: vc_height, vc_moist_pressure, vc_dry_pressure
       use air_composition, only: wv_idx
-      use physconst,       only: gravit, latvap, latice
+      use air_composition, only: dry_air_species_num
+      use physconst,       only: rga, latvap, latice
 
       ! Dummy arguments
       ! tracer: tracer mixing ratio
+      !
+      ! note - if pdeldry passed to subroutine then tracer mixing ratio must be dry
       real(kind_phys), intent(in)            :: tracer(:,:,:)
+      logical, intent(in)             :: moist_mixing_ratio
       ! pdel: pressure level thickness
-      real(kind_phys), intent(in)            :: pdel(:,:)
-      ! cp_or_cv: dry air heat capacity under constant pressure or
-      !           constant volume (depends on vcoord)
+      real(kind_phys), intent(in)            :: pdel_in(:,:)
+      ! cp_or_cv: air heat capacity under constant pressure or
+      !           constant volume (depends on energy formula)
       real(kind_phys), intent(in)            :: cp_or_cv(:,:)
       real(kind_phys), intent(in)            :: U(:,:)
       real(kind_phys), intent(in)            :: V(:,:)
       real(kind_phys), intent(in)            :: T(:,:)
-      integer,         intent(in)            :: vcoord ! vertical coordinate
-      real(kind_phys), intent(in),  optional :: ps(:)
+      integer,         intent(in)            :: vcoord !REMOVECAM - vcoord or energy formula to use
```

Comment:
I assume "REMOVECAM" is a reminder for future renaming in cam_development/SIMA to make code clearer in terms of variable names?